src/vs/workbench/contrib/chat/browser/chatViewPane.ts

```diff
@@ -66,6 +66,11 @@ export class ChatViewPane extends ViewPane implements IChatViewPane {
 		// View state for the ViewPane is currently global per-provider basically, but some other strictly per-model state will require a separate memento.
 		this.memento = new Memento('interactive-session-view-' + this.chatViewOptions.providerId, this.storageService);
 		this.viewState = this.memento.getMemento(StorageScope.WORKSPACE, StorageTarget.MACHINE) as IViewPaneState;
+		this._register(this.chatService.onDidRegisterProvider(({ providerId }) => {
+			if (providerId === this.chatViewOptions.providerId && !this._widget?.viewModel) {
+				this.updateModel();
```

Comment:
Won't this still not use the persisted session? I think you need to do all of this

https://github.com/microsoft/vscode/blob/9e3c26b76e78092e5243e3484f810cb3347b8861/src/vs/workbench/contrib/chat/browser/chatViewPane.ts#L111-L120